_dev/src/ts/components/ComponentAbstract.ts

```diff
@@ -0,0 +1,13 @@
+export default abstract class ComponentAbstract {
+  public element: HTMLElement;
+
+  public constructor(element: HTMLElement) {
+    this.element = element;
+  }
+
+  protected queryElement = <T extends HTMLElement>(selector: string, errorMessage: string): T => {
+    const element = (this.element.querySelector(selector) as T) ?? document.querySelector(selector);
+    if (!element) throw new Error(errorMessage);
```

Comment:
```suggestion
    if (!element) {
       throw new Error(errorMessage);
     }
```

to gain readability and be consistent with the rest of the code